pkg/minikube/registry/drvs/hyperv/vswitch.go

```diff
@@ -0,0 +1,212 @@
+	if err != nil {
+		return nil, err
+	}
+
+	return adapters, nil
+}
+
+func getVMSwitch(condition string) ([]vmSwitch, error) {
+	cmd := []string{"Hyper-V\\Get-VMSwitch"}
+	if condition != "" {
+		cmd = append(cmd, fmt.Sprintf("Where-Object {%s}", condition))
+	}
+	cmd = append(cmd, "Select-Object -Property Name, NetAdapterInterfaceGuid")
+	stdout, err := cmdOut(fmt.Sprintf("ConvertTo-Json @(%s)", strings.Join(cmd, " | ")))
+	if err != nil {
+		return nil, err
+	}
+
+	var vmSwitches []vmSwitch
+	err = json.Unmarshal([]byte(strings.TrimSpace(stdout)), &vmSwitches)
+	if err != nil {
+		return nil, err
+	}
+
+	return vmSwitches, nil
+}
+
+func findConnectedVMSwitch(adapterGUID string) (string, error) {
+	foundSwitches, err := getVMSwitch(fmt.Sprintf("($_.SwitchType -eq 2) -And ($_.NetAdapterInterfaceGuid -contains \"%s\")", adapterGUID))
+	if err != nil {
+		return "", err
+	}
+
+	if len(foundSwitches) > 0 {
+		return foundSwitches[0].Name, nil
+	}
+
+	return "", nil
+}
+
+func getOrderedAdapters() ([]netAdapter, error) {
+	// look for "Up" adapters and prefer physical LAN over other options
+	lanAdapters, err := getNetAdapters(true, "($_.Status -eq \"Up\") -And ($_.PhysicalMediaType -like \"*802.3*\")")
+	if err != nil {
+		return nil, err
+	}
+
+	upAdapters, err := getNetAdapters(true, "$_.Status -eq \"Up\"")
+	if err != nil {
+		return nil, err
+	}
+
+	var orderedAdapters []netAdapter
+	adapterGuids := map[string]interface{}{}
+
+	// all adapters will be checked in the following order:
+	//  1. Connected physical LAN adapters
+	//  2. Any other connected adapters
+	for _, adapter := range lanAdapters {
+		if _, ok := adapterGuids[adapter.InterfaceGUID]; !ok {
+			adapterGuids[adapter.InterfaceGUID] = nil
+			orderedAdapters = append(orderedAdapters, adapter)
+		}
+	}
+
+	for _, adapter := range upAdapters {
+		if _, ok := adapterGuids[adapter.InterfaceGUID]; !ok {
+			adapterGuids[adapter.InterfaceGUID] = nil
+			orderedAdapters = append(orderedAdapters, adapter)
+		}
+	}
+
+	return orderedAdapters, nil
+}
+
+func createVMSwitch(switchName string, adapter netAdapter) error {
+	err := cmd(fmt.Sprintf("Hyper-V\\New-VMSwitch -Name \"%s\" -NetAdapterInterfaceDescription \"%s\"", switchName, adapter.InterfaceDescription))
+	if err != nil {
+		return errors.Wrapf(err, "failed to create VM switch %s with adapter %s", switchName, adapter.InterfaceGUID)
+	}
+
+	return nil
+}
+
+func chooseSwitch(adapterName string) (string, error) {
+	var adapter netAdapter
+	if adapterName != "" {
+		foundAdapters, err := getNetAdapters(false, fmt.Sprintf("($_.InterfaceDescription -eq \"%s\")", adapterName))
+		if err != nil {
+			return "", err
+		}
+
+		if len(foundAdapters) == 0 {
+			return "", errors.Errorf("adapter %s not found", adapterName)
+		}
+
+		adapter = foundAdapters[0]
+		foundSwitch, err := findConnectedVMSwitch(adapter.InterfaceGUID)
+		if err != nil {
+			return "", err
+		}
+
+		if foundSwitch != "" {
+			return foundSwitch, nil
+		}
+
```

Comment:
Can we return here to remove the requirement of having an else case?

Reply:
Done